src/Microsoft.Diagnostics.ExtensionCommands/ThreadPoolCommand.cs

```diff
@@ -70,8 +80,8 @@ public override void Invoke()
 
                 // We will assume that if UsePortableThreadPoolForIO field is deleted from ThreadPool then we are always
                 // using C# version.
-                bool usingPortableCompletionPorts = threadPool.Portable && (usePortableIOField is null || usePortableIOField.Read<bool>(usePortableIOField.Type.Module.AppDomain));
-                if (!usingPortableCompletionPorts)
+                bool usingPortableCompletionPorts = threadPool.UsingPortableThreadPool && (usePortableIOField is null || usePortableIOField.Read<bool>(usePortableIOField.Type.Module.AppDomain));
```

Comment:
If the `UsePortableThreadPoolForIO` field doesn't exist, what should the behavior be?  That's what the original code says.  IE in .Net 6, before that field was added, did we use portable completion ports or no?

That's what the original code is getting at.

Reply:
The info shown when this variable is false is info about the IO completion thread pool. On Windows:
- The IO completion thread pool exists in .NET 7 and earlier
- It is the only option in .NET 6 and below. The `UsePortableThreadPoolForIO` field doesn't exist.
- In .NET 7, the `UsePortableThreadPoolForIO` field exists and is true by default, in which case the IO completion thread pool is not used, but that can be changed through config
- In .NET 8, the `UsePortableThreadPoolForIO` field doesn't exist and the IO completion thread pool doesn't exist. However, in .NET 8, `GetThreadpoolData` returns `E_NOTIMPL`.

On Unixes, there has never been an IO completion thread pool, so this info shouldn't be shown ideally.

Currently it's difficult to distinguish the .NET 8 case from the .NET 6 case since `UsePortableThreadPoolForIO` doesn't exist in both cases. Maybe https://github.com/microsoft/clrmd/pull/1175 could expose whether the legacy thread pool data is available and that could be included to the condition?

Reply:
Feel free to make any changes in ClrMD that make sense for this scenario.

Reply:
Maybe something like this would work:
```c#
bool usingIOCompletionThreadPool =
    threadPool.HasLegacyData &&
    (
        usePortableIOField is null ||
        !usePortableIOField.Read<bool>(usePortableIOField.Type.Module.AppDomain)
    );
if (usingIOCompletionThreadPool)
{
    // Write info about IO completion thread pool
}
```

It would still print some unnecessary info for Unixes until .NET 8, that's probably ok.